src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -329,7 +345,15 @@ private static boolean isStarImport(DetailAST importAst) {
      */
     private void checkClassName(DetailAST ast) {
         final DetailAST type = ast.findFirstToken(TokenTypes.TYPE);
```

Comment:
http://checkstyle.sourceforge.net/config_coding.html#IllegalType
Please see properties `ignoredMethodNames` and `memberModifiers` - they have to be applied when any `TYPE` is used, that is why I think we check `TYPE` only at this point and do not add it to acceptable tokens. The same goes for generics. So please:
1) remove new `TokenTypes` from `getAcceptableTokens` and modify `checkClassName` method
2) add new UTs: generics + `ignoredMethodNames` and generics + `memberModifiers`

Reply:
fixed, but generics not checked for `CLASS_DEF`, because in [documentation](http://checkstyle.sourceforge.net/config_coding.html#IllegalType) is description:
`Checks that particular classes are never used as types in variable declarations, return values or parameters.`. I don't know if I did the right thing.

Reply:
I think, need return TokenTypes, because it simplest way for resolve current issue.